### PR TITLE
transpose kernel in vec_ops and rust binding

### PR DESCRIPTION
**Pull request by vladfdp**
## Describe the changes

This PR adds an extern C link to the transpose kernel, now in vec_ops.cu.
Also Rust binding, and I updated the test check_ntt_batch to use the new transpose function.
The test passes.

## Linked Issues

Resolves #

### CODE REVIEW

**Review by ChickenLover — COMMENTED** [review]
Congrats with your first PR @vladfdp! Excellent work 

I left some minor comments. Also it's generally a good idea to add a test for any new functionally. Take a look at [this](https://github.com/ingonyama-zk/icicle/blob/feat/vlad/transpose-api/wrappers/rust/icicle-core/src/vec_ops/tests.rs#L4) test for example. You can add something like this for the transpose function

File: icicle/utils/vec_ops.cu [hunk 1]
```diff
@@ -101,6 +109,35 @@ namespace vec_ops {
     return VecOp<E, SubKernel>(vec_a, vec_b, n, config, result);
   }
 
+  template <typename E>
+  cudaError_t transpose_batch(
```
[the remark this answers is not in the document]
Comment (vladfdp) [reply icicle/utils/vec_ops.cu:113]: Yes it's just that the kernel I was asked to move in verification.cu was called transpose_batch (probably due to the context it was used in) so I kept the name. What better name would you suggest ?

[the remark this answers is not in the document]
Comment (ChickenLover) [reply icicle/utils/vec_ops.cu:113]: I think `transpose_matrix` will be fine. It's the name you use for the Rust wrapper anyway:)

File: icicle/utils/vec_ops.cu [hunk 2]
```diff
@@ -146,4 +185,20 @@ namespace vec_ops {
     return Sub<curve_config::scalar_t>(vec_a, vec_b, n, config, result);
   }
 
+  /**
+   * Extern version of transpose_batch function with the template parameter
+   * `E` being the [scalar field](@ref scalar_t) of the curve given by `-DCURVE` env variable during build.
+   * @return `cudaSuccess` if the execution was successful and an error code otherwise.
+   */
+  extern "C" cudaError_t CONCAT_EXPAND(CURVE, TransposeBatch)(
+    curve_config::scalar_t* input,
```
Comment (yshekel) [inline icicle/utils/vec_ops.cu:194]: better make the input const pointer (const scalar*). This is also true for the kernels.
In some cases (for example polynomials) the internal memory can only be accessed by const so this way means this API cannot be used.

**Review by LeonHibnik — APPROVED** [review]

**Review by ChickenLover — APPROVED** [review]

**Pull request merged**